server/rpc/interceptors/interceptors.go

```diff
@@ -257,6 +260,39 @@ func quotaStreamServerInterceptor(env environment.Env) grpc.StreamServerIntercep
 	}
 }
 
+func alertOnPanic() {
+	buf := make([]byte, 1<<20)
+	n := runtime.Stack(buf, true)
+	alert.UnexpectedEvent("recovered_panic", buf[:n])
+}
+
+func unaryRecoveryInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (rsp interface{}, err error) {
+		defer func() {
+			if panicErr := recover(); panicErr != nil {
+				rsp = nil
+				err = status.InternalError("A panic occurred")
+				alertOnPanic()
+			}
+		}()
+		rsp, err = handler(ctx, req)
+		return
```

Comment:
Is this the, uh, idiomatic way to write these statements (instead of just doing `return handler(ctx, req)` (and `return handler(srv, stream)` below)?  Sincerely just a question, just learning what I should be doing to fit in.